File: src/main/java/no/nav/bidrag/beregn/kostnadsberegnetbidrag/beregning/KostnadsberegnetBidragBeregning.java
package no.nav.bidrag.beregn.kostnadsberegnetbidrag.beregning;

import no.nav.bidrag.beregn.kostnadsberegnetbidrag.bo.GrunnlagBeregning;
import no.nav.bidrag.beregn.kostnadsberegnetbidrag.bo.ResultatBeregning;

public interface KostnadsberegnetBidragBeregning {

  ResultatBeregning beregn(GrunnlagBeregning grunnlagBeregning);

  static KostnadsberegnetBidragBeregning getInstance(){
    return new KostnadsberegnetBidragBeregningImpl();
  }
}

File: src/main/kotlin/no/nav/bidrag/beregn/kostnadsberegnetbidrag/beregning/KostnadsberegnetBidragBeregning.kt
package no.nav.bidrag.beregn.kostnadsberegnetbidrag.beregning

import no.nav.bidrag.beregn.kostnadsberegnetbidrag.bo.GrunnlagBeregning
import no.nav.bidrag.beregn.kostnadsberegnetbidrag.bo.ResultatBeregning

fun interface KostnadsberegnetBidragBeregning {
    fun beregn(grunnlag: GrunnlagBeregning): ResultatBeregning

    companion object {
        fun getInstance(): KostnadsberegnetBidragBeregning {
            return KostnadsberegnetBidragBeregningImpl()
        }
    }
}
